### PR TITLE
bump knex to v3 and better-sqlite3 to v9

### PR DESCRIPTION
**Pull request by freben**
After node 16 went EOL, both `knex` and `better-sqlite3` carved out new major versions that cut the ties with old node versions. It's time to get on these trains so we keep getting updates.

The corresponding little `pg` bump is not of particular importance; we already had resolved to that version in our lockfile. They don't have a corresponding major bump (yet?). But given the recent CI problems etc I just wanted to make sure everyone is on recent versions of all the database libraries.

The libraries don't seem to have tried to fit in any other major changes, so this _should_ be a relatively low risk change, but let's still do it after the mainline :) In particular the sqlite version is very recent too.

### CODE REVIEW

**Comment by freben** [comment]
Hm, minor type error with `connect-session-knex`. Reported upstream: https://github.com/gx0r/connect-session-knex/issues/103

File: yarn.lock [hunk 1]
```diff
@@ -31167,7 +31167,7 @@ __metadata:
   languageName: node
   linkType: hard
 
-"knex@npm:^2.0.0, knex@npm:^2.3.0, knex@npm:^2.4.2":
+"knex@npm:^2.3.0":
```
Comment (tudi2d) [inline yarn.lock:31170]: A bit suprised about this usage of `knex `^2` version. Is it still used somewhere/ a peer dependency?

**Review by tudi2d — APPROVED** [review]
Looks good - bit suprised about the `knex` v2 refernce staying in the `yarn.lock`. Is it maybe still used somewhere where we can't bump it or is there somewhere a dependency we missed? 😕

File: yarn.lock (hunk 1 above)
Comment (freben) [reply yarn.lock:31170]: Yep that's right, the one remaining offender here is the `connect-session-knex` library mentioned above, which indeed is having type issues after upgrading the repo as we pass in knex3 clients to that knex2 library.

Comment (tudi2d) [reply yarn.lock:31170]: Ok - let's not blocking for me than. :shipit:.

Comment (freben) [reply yarn.lock]: Now it's gone! The makers of that library released a new version that builds with knex3, so I bumped that too

**Comment by awanlin** [comment]
Thanks for this @freben 🚀

**Pull request merged**